CMakeLists.txt

```diff
@@ -467,32 +471,31 @@ if(ACTS_BUILD_PLUGIN_GEANT4)
     find_package(Geant4 ${_acts_geant4_version} REQUIRED CONFIG COMPONENTS gdml)
 endif()
 
-if(ACTS_BUILD_PLUGIN_TRACCC)
-    if(ACTS_USE_SYSTEM_ALGEBRAPLUGINS)
-        find_package(algebra-plugins ${_acts_algebraplugins_version} REQUIRED)
-    else()
-        add_subdirectory(thirdparty/algebra-plugins)
-    endif()
-
-    if(ACTS_USE_SYSTEM_DETRAY)
-        find_package(detray ${_acts_detray_version} REQUIRED CONFIG)
+# Convenience macro to simplify the check if we want to setup, and if
+# so, if we want to find it in the system or build internally
+macro(conditional_setup _var _package _path)
+    if(ACTS_SETUP_${_var})
+        if(ACTS_USE_SYSTEM_${_var})
+            message(VERBOSE "Use system ${_package}")
```

Comment:
Careful with all these `VERBOSE` messages! These have been introduced in CMake 3.15.

https://cmake.org/cmake/help/latest/command/message.html#general-messages

But Acts only seems to require CMake 3.14 at the moment.

https://github.com/acts-project/acts/blob/main/CMakeLists.txt#L1

I have to admit I didn't even know that `VERBOSE` messages existed. And I wonder what the use-case really is. Since debugging CMake can be done without the help of debugging messages. (CMake can be put in a trace mode where you can check exactly what's happening.)

So on first order I'd just print the meaningful / useful stuff as `STATUS`, and not bother about any `VERBOSE` messages.

Reply:
yeah we can remove this... Good to know, I didn't new about trace mode actually